crates/control/src/motion/joint_command_sender.rs

```diff
@@ -95,12 +96,20 @@ impl JointCommandSender {
                 ),
             ),
         };
+
+        // The actuators uses the raw sensor data (not corrected like current_positions) in their feedback loops,
+        // thus the compensation is required to make them reach the actual desired position.
+        let compensated_positions = positions + *context.joint_calibration_offsets;
+
         context
             .hardware_interface
-            .write_to_actuators(positions, stiffnesses, *context.leds)
+            .write_to_actuators(compensated_positions, stiffnesses, *context.leds)
             .wrap_err("failed to write to actuators")?;
 
-        context.positions.fill_if_subscribed(|| positions);
+        context
+            .positions
+            .fill_if_subscribed(|| compensated_positions);
+
```

Comment:
This seems wrong to me.
I would argue that writing "compensated positions" into an additional output called "positions" is confusing.
Perhaps add a second output instead and output both the requested and compensated positions?

Reply:
Yes you are right, I was confused a bit and corrected the others but not this.

Compensated position is what the hardware thinks it has to reach while the position is the actual position we need to reach.

I'll add an additional output in case it helps.

Reply:
Fixed